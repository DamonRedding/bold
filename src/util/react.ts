import { MutableRefObject, Ref } from 'react'

export function composeRefs<T>(...refs: Array<Ref<T>>) {
    return (value: T) => refs.forEach(ref => setRef(ref, value))
}

export function setRef<T>(ref: Ref<T>, value: any) {
    if (typeof ref === 'function') {
        ref(value)
    } else if (ref) {
        (ref as MutableRefObject<T>).current = value
    }
}

/**
 * Returns the next sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getNextSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.nextElementSibling) {
        current = current.nextElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}

/**
 * Returns the previous sibling of `element` that matches the `predicate`.
 * Returns `null` if not found.
 */
export function getPreviousSibling(element: Element, predicate: (sibling: Element) => boolean) {
    let current: Element = element

    while (current.previousElementSibling) {
        current = current.previousElementSibling

        if (predicate(current)) {
            return current
        }
    }

    return null
}
