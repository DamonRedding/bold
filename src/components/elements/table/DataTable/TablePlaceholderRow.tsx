import * as React from 'react'

import { withStyles, WithStylesProps } from '../../../../styles'
import { TableCell, TableRow } from '../Table'

export interface TablePlaceholderRowProps extends WithStylesProps {
    colSpan: number
    message?: string
}

@withStyles
export class TablePlaceholderRow extends React.PureComponent<TablePlaceholderRowProps> {
    static defaultProps: Partial<TablePlaceholderRowProps> = {
        message: 'Nenhum registro encontrado',
    }

    render() {
        const { theme, colSpan, message } = this.props
        const styles = {
            cell: {
                color: theme.color.gray40,
                fontStyle: 'italic',
                textAlign: 'center',
            },
        }
        return (
            <TableRow>
                <TableCell colSpan={colSpan} styles={styles.cell}>
                    {message}
                </TableCell>
            </TableRow>
        )
    }
}
