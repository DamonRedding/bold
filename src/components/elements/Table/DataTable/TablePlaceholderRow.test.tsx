import { render } from 'enzyme'
import * as React from 'react'

import { withTheme } from '../../../../test'

import { TablePlaceholderRow } from './TablePlaceholderRow'

it('should render correctly', () => {
    expect(render(withTheme(
        <TablePlaceholderRow colSpan={3} />
    ))).toMatchSnapshot()
    expect(render(withTheme(
        <TablePlaceholderRow colSpan={2} message='No results' />
    ))).toMatchSnapshot()
})
