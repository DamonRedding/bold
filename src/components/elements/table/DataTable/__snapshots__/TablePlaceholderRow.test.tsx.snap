// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
.emotion-0 {
  text-align: left;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #e6e6e6;
  color: #666666;
  font-style: italic;
  text-align: center;
}

<tr
  class="emotion-1"
>
  <td
    class="emotion-0"
    colspan="3"
  >
    Nenhum registro encontrado
  </td>
</tr>
`;

exports[`should render correctly 2`] = `
.emotion-0 {
  text-align: left;
  padding: 0.5rem 1rem;
  border-bottom: 1px solid #e6e6e6;
  color: #666666;
  font-style: italic;
  text-align: center;
}

<tr
  class="emotion-1"
>
  <td
    class="emotion-0"
    colspan="2"
  >
    No results
  </td>
</tr>
`;
